### ISSUE TITLE
Doesn't seem to be working

### ISSUE THREAD

**Issue by zkvsky** [prefiks/crown-controller#1]
I'm trying it with firefox [ not -bin  and yes I've changed the config.yaml ] plus some other apps and it appears to do nothing besides disabling volume keys.
How does the app detection work? I'm running KDE/Plasma

**Comment by prefiks** [comment]
Hello,

It tries to detect active app by reading window NET_WM_PID property  (you could try to run `xprop | grep NET_WM_PID` to get value for window) that should contain process pid, and then by checking to what binary `/proc/[PID]/exe` points to. I only tested it on gnome, and it works only on X11 not Wayland.

**Comment by prefiks** [comment]
And you can check what binary that program detect from shell with `readling /proc/[PID]/exe`, it will try to match whole path or if that fail last part of path.

**Comment by zkvsky** [comment]
I'm using X11 not wayland, kde has inbuilt app/windows detection tools and the process is indeed just firefox, readlink reports the same, I've tried several apps already though and it doesn't work with any.
If it doesn't work under KDE/Plasma just state so in the docs.

**Comment by prefiks** [comment]
I didn't test it with kde, so didn't really know if it work with it or not, do you see any output in shell when you run that program and try to do any crown command? Do command set in global section (like ctrl + rotation if you have that part of config) generate any output?

**Comment by zkvsky** [comment]
I have the global part, it doesn't work, just disables the crown altogether, no output
Which firmware version are you on?

**Comment by prefiks** [comment]
Ok, then maybe my parser for events that are generated doesn't recognize your what is produced in your system. If you are willing to test it i could prepare binary that will try to print raw data to standard output so i would be able what differs between our outputs.

**Comment by zkvsky** [comment]
Sure, go ahead

**Comment by prefiks** [comment]
Here it is, it should write to standard output informations about app switch/crown events

[program.zip](https://github.com/prefiks/crown-controller/files/5279484/program.zip)

**Comment by zkvsky** [comment]
here you go:
<details><summary>log</summary>
<p>


```python
./crown-controller 
Crown events: [11, 2, 12, 21, 2, 2, 32, 28, 19, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] Unknown
Crown events: [1, 0, 0, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
App switch: 31066 /usr/lib64/firefox/firefox
Mode events: EnableRatchet
App switch: 31066 /usr/lib64/firefox/firefox
Mode events: EnableRatchet
Crown events: [11, 2, 12, 21, 2, 2, 32, 28, 19, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] Unknown
Crown events: [11, 2, 12, 21, 2, 2, 32, 28, 19, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] Unknown
App switch: 30952 /usr/bin/konsole
Mode events: EnableRatchet
App switch: 30952 /usr/bin/konsole
Mode events: EnableRatchet
Crown events: [11, 2, 12, 21, 2, 2, 32, 28, 19, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] Unknown
Crown events: [11, 2, 12, 21, 2, 2, 32, 28, 19, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] Unknown
Crown events: [11, 2, 12, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] Unknown
Crown events: [11, 2, 12, 0, 0, 0, 0, 2, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] Unknown
Crown events: [11, 2, 12, 0, 0, 0, 0, 2, 3, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] Unknown
Crown events: [11, 2, 12, 0, 0, 0, 0, 2, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] Unknown
Crown events: [11, 2, 12, 0, 0, 0, 0, 2, 3, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] Unknown
Crown events: [11, 2, 12, 0, 0, 0, 0, 2, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] Unknown
Crown events: [11, 2, 12, 0, 0, 0, 0, 2, 3, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] Unknown
App switch: 31066 /usr/lib64/firefox/firefox
Mode events: EnableRatchet
App switch: 31066 /usr/lib64/firefox/firefox
Mode events: EnableRatchet
Crown events: [11, 2, 12, 21, 2, 2, 32, 28, 19, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] Unknown
Crown events: [11, 2, 12, 21, 2, 2, 32, 28, 19, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] Unknown
Crown events: [11, 2, 12, 0, 0, 0, 0, 2, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] Unknown
Crown events: [11, 2, 12, 0, 1, 1, 0, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] Unknown
Crown events: [11, 2, 12, 0, 2, 1, 0, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ae] Unknown
Crown events: [11, 2, 12, 0, 2, 2, 1, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ae] Unknown
Crown events: [11, 2, 12, 0, 2, 1, 0, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ae] Unknown
Crown events: [11, 2, 12, 0, 2, 1, 0, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ae] Unknown
Crown events: [11, 2, 12, 0, 2, 1, 0, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 2a] Unknown
Crown events: [11, 2, 12, 0, 2, 1, 0, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ae] Unknown
Crown events: [11, 2, 12, 0, 2, 1, 0, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 57] Unknown
Crown events: [11, 2, 12, 0, 2, 1, 1, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 57] Unknown
Crown events: [11, 2, 12, 0, 2, 2, 0, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ae] Unknown
Crown events: [11, 2, 12, 0, 2, 1, 0, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 21] Unknown
Crown events: [11, 2, 12, 0, 2, ff, 0, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, ff, e4] Unknown
Crown events: [11, 2, 12, 0, 2, ff, 0, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, ff, c8] Unknown
Crown events: [11, 2, 12, 0, 2, ff, 0, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, ff, 52] Unknown
Crown events: [11, 2, 12, 0, 2, ff, ff, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, ff, a9] Unknown
Crown events: [11, 2, 12, 0, 2, fe, 0, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, ff, 52] Unknown
Crown events: [11, 2, 12, 0, 2, ff, 0, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, ff, 52] Unknown
Crown events: [11, 2, 12, 0, 2, fe, 0, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, ff, 52] Unknown
Crown events: [11, 2, 12, 0, 2, ff, 0, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, ff, 52] Unknown
Crown events: [11, 2, 12, 0, 2, fe, ff, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, fe, a1] Unknown
Crown events: [11, 2, 12, 0, 2, ff, 0, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, ff, a9] Unknown
Crown events: [11, 2, 12, 0, 2, ff, 0, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, ff, ea] Unknown
Crown events: [11, 2, 12, 0, 2, fe, 0, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, ff, 8d] Unknown
Crown events: [11, 2, 12, 0, 2, ff, ff, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, ff, 52] Unknown
Crown events: [11, 2, 12, 0, 2, fe, 0, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, fe, a1] Unknown
Crown events: [11, 2, 12, 0, 2, 1, 0, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 10] Unknown
Crown events: [11, 2, 12, 0, 2, 3, 1, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 2, d] Unknown
Crown events: [11, 2, 12, 0, 2, 2, 0, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 5f] Unknown
Crown events: [11, 2, 12, 0, 2, 1, 0, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 2a] Unknown
Crown events: [11, 2, 12, 0, 2, 1, 0, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ae] Unknown
Crown events: [11, 2, 12, 0, 2, 2, 1, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 5f] Unknown
Crown events: [11, 2, 12, 0, 2, 2, 0, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 5f] Unknown
Crown events: [11, 2, 12, 0, 2, 1, 0, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 2a] Unknown
Crown events: [11, 2, 12, 0, 2, 1, 0, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 38] Unknown
Crown events: [11, 2, 12, 0, 2, ff, 0, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, ff, f2] Unknown
Crown events: [11, 2, 12, 0, 2, ff, 0, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, ff, a9] Unknown
Crown events: [11, 2, 12, 0, 2, ff, 0, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, ff, df] Unknown
Crown events: [11, 2, 12, 0, 2, fe, 0, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, fe, a1] Unknown
Crown events: [11, 2, 12, 0, 2, fd, ff, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, fd, f3] Unknown
Crown events: [11, 2, 12, 0, 2, ff, 0, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, ff, a9] Unknown
Crown events: [11, 2, 12, 0, 2, ff, 0, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, ff, 52] Unknown
Crown events: [11, 2, 12, 0, 2, ff, 0, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, ff, 52] Unknown
Crown events: [11, 2, 12, 0, 2, ff, ff, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, ff, 52] Unknown
Crown events: [11, 2, 12, 0, 2, fe, 0, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, fe, a1] Unknown
Crown events: [11, 2, 12, 0, 2, 0, 0, 2, 3, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] Unknown
Crown events: [11, 2, 12, 0, 3, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] Unknown
Crown events: [11, 2, 12, 0, 0, 0, 0, 2, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] Unknown
Crown events: [11, 2, 12, 0, 0, 0, 0, 2, 3, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] Unknown
Crown events: [11, 2, 12, 0, 0, 0, 0, 2, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] Unknown
Crown events: [11, 2, 12, 0, 1, 1, 0, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] Unknown
Crown events: [11, 2, 12, 0, 2, 1, 0, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ae] Unknown
Crown events: [11, 2, 12, 0, 2, 3, 1, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 2, d] Unknown
Crown events: [11, 2, 12, 0, 2, 1, 0, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 38] Unknown
Crown events: [11, 2, 12, 0, 2, 1, 0, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 57] Unknown
Crown events: [11, 2, 12, 0, 2, 1, 0, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 57] Unknown
Crown events: [11, 2, 12, 0, 2, ff, 0, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, ff, f5] Unknown
Crown events: [11, 2, 12, 0, 2, fe, 0, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, ff, 52] Unknown
Crown events: [11, 2, 12, 0, 2, ff, 0, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, ff, ea] Unknown
Crown events: [11, 2, 12, 0, 2, fe, 0, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, fe, b7] Unknown
Crown events: [11, 2, 12, 0, 2, fd, ff, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, fd, f3] Unknown
Crown events: [11, 2, 12, 0, 2, 0, 0, 2, 3, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] Unknown
Crown events: [11, 2, 12, 0, 3, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] Unknown
Crown events: [11, 2, 12, 0, 0, 0, 0, 2, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] Unknown
Crown events: [11, 2, 12, 0, 0, 0, 0, 2, 2, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0] Unknown
Crown events: [11, 2, 12, 0, 0, 0, 0, 2, 2, 0, 3, 0, 0, 0, 0, 0, 0, 0, 0, 0] Unknown
Crown events: [11, 2, 12, 0, 1, 2, 0, 2, 2, 0, 4, 0, 0, 0, 0, 0, 0, 0, 0, 0] Unknown
Crown events: [11, 2, 12, 0, 2, 2, 1, 2, 2, 0, 4, 0, 0, 0, 0, 0, 0, 0, 1, 5f] Unknown
Crown events: [11, 2, 12, 0, 2, 2, 0, 2, 2, 0, 4, 0, 0, 0, 0, 0, 0, 0, 1, 5f] Unknown
Crown events: [11, 2, 12, 0, 2, 1, 0, 2, 2, 0, 4, 0, 0, 0, 0, 0, 0, 0, 0, 2a] Unknown
Crown events: [11, 2, 12, 0, 2, 2, 0, 2, 2, 0, 4, 0, 0, 0, 0, 0, 0, 0, 0, ae] Unknown
Crown events: [11, 2, 12, 0, 2, 2, 1, 2, 2, 0, 4, 0, 0, 0, 0, 0, 0, 0, 1, 5f] Unknown
Crown events: [11, 2, 12, 0, 2, 1, 0, 2, 2, 0, 4, 0, 0, 0, 0, 0, 0, 0, 0, ae] Unknown
Crown events: [11, 2, 12, 0, 2, ff, 0, 2, 2, 0, 4, 0, 0, 0, 0, 0, 0, 0, ff, f8] Unknown
Crown events: [11, 2, 12, 0, 2, ff, 0, 2, 2, 0, 4, 0, 0, 0, 0, 0, 0, 0, ff, 52] Unknown
Crown events: [11, 2, 12, 0, 2, fe, ff, 2, 2, 0, 4, 0, 0, 0, 0, 0, 0, 0, fe, a1] Unknown
Crown events: [11, 2, 12, 0, 2, fe, 0, 2, 2, 0, 4, 0, 0, 0, 0, 0, 0, 0, fe, a1] Unknown
Crown events: [11, 2, 12, 0, 2, ff, 0, 2, 2, 0, 4, 0, 0, 0, 0, 0, 0, 0, ff, a9] Unknown
Crown events: [11, 2, 12, 0, 2, ff, 0, 2, 2, 0, 4, 0, 0, 0, 0, 0, 0, 0, ff, df] Unknown
Crown events: [11, 2, 12, 0, 2, ff, 0, 2, 2, 0, 4, 0, 0, 0, 0, 0, 0, 0, ff, 52] Unknown
Crown events: [11, 2, 12, 0, 2, fe, ff, 2, 2, 0, 4, 0, 0, 0, 0, 0, 0, 0, fe, a1] Unknown
Crown events: [11, 2, 12, 0, 2, ff, 0, 2, 2, 0, 4, 0, 0, 0, 0, 0, 0, 0, ff, 52] Unknown
Crown events: [11, 2, 12, 0, 2, 3, 0, 2, 2, 0, 4, 0, 0, 0, 0, 0, 0, 0, 0, 19] Unknown
Crown events: [11, 2, 12, 0, 2, 2, 1, 2, 2, 0, 4, 0, 0, 0, 0, 0, 0, 0, 1, 5f] Unknown
Crown events: [11, 2, 12, 0, 2, 1, 0, 2, 2, 0, 4, 0, 0, 0, 0, 0, 0, 0, 0, ae] Unknown
Crown events: [11, 2, 12, 0, 2, 2, 0, 2, 2, 0, 4, 0, 0, 0, 0, 0, 0, 0, 0, 57] Unknown
Crown events: [11, 2, 12, 0, 2, 1, 0, 2, 2, 0, 4, 0, 0, 0, 0, 0, 0, 0, 0, ae] Unknown
Crown events: [11, 2, 12, 0, 2, 2, 1, 2, 2, 0, 4, 0, 0, 0, 0, 0, 0, 0, 1, 5f] Unknown
Crown events: [11, 2, 12, 0, 2, 1, 0, 2, 2, 0, 4, 0, 0, 0, 0, 0, 0, 0, 0, ae] Unknown
Crown events: [11, 2, 12, 0, 2, ff, 0, 2, 2, 0, 4, 0, 0, 0, 0, 0, 0, 0, ff, f8] Unknown
Crown events: [11, 2, 12, 0, 2, ff, 0, 2, 2, 0, 4, 0, 0, 0, 0, 0, 0, 0, ff, 52] Unknown
Crown events: [11, 2, 12, 0, 2, fe, ff, 2, 2, 0, 4, 0, 0, 0, 0, 0, 0, 0, fe, a1] Unknown
Crown events: [11, 2, 12, 0, 2, fe, 0, 2, 2, 0, 4, 0, 0, 0, 0, 0, 0, 0, fe, a1] Unknown
Crown events: [11, 2, 12, 0, 2, ff, 0, 2, 2, 0, 4, 0, 0, 0, 0, 0, 0, 0, ff, a9] Unknown
Crown events: [11, 2, 12, 0, 2, ff, 0, 2, 2, 0, 4, 0, 0, 0, 0, 0, 0, 0, ff, 52] Unknown
Crown events: [11, 2, 12, 0, 2, fd, ff, 2, 2, 0, 4, 0, 0, 0, 0, 0, 0, 0, fd, f3] Unknown
Crown events: [11, 2, 12, 0, 2, ff, 0, 2, 2, 0, 4, 0, 0, 0, 0, 0, 0, 0, ff, 52] Unknown
Crown events: [11, 2, 12, 0, 2, 0, 0, 2, 2, 0, 5, 0, 0, 0, 0, 0, 0, 0, 0, 0] Unknown
Crown events: [11, 2, 12, 0, 2, 0, 0, 2, 3, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] Unknown
Crown events: [11, 2, 12, 0, 3, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] Unknown
Mode events: EnableRatchet
App switch: 30952 /usr/bin/konsole
Mode events: EnableRatchet
Crown events: [11, 2, 12, 21, 2, 2, 32, 28, 19, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] Unknown
Crown events: [11, 2, 12, 21, 2, 2, 32, 28, 19, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] Unknown
App switch: 31579 /usr/lib64/firefox/firefox
Mode events: EnableRatchet
App switch: 31579 /usr/lib64/firefox/firefox
Mode events: EnableRatchet
Crown events: [11, 2, 12, 21, 2, 2, 32, 28, 19, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] Unknown
Crown events: [11, 2, 12, 21, 2, 2, 32, 28, 19, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] Unknown
Crown events: [1, 0, b, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 0, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 8, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 0, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 15, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 0, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 8, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 0, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 2c, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 0, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 1c, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 0, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 12, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 0, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 18, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 0, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 2c, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 0, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, a, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 0, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 12, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 0, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, a, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 0, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, c, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 0, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 17, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 0, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, b, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 0, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 18, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 0, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 5, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 0, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 2c, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 0, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 6, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 0, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 12, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 0, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, f, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 0, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, f, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 0, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 4, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 0, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 51, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 0, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 51, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 0, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 51, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 0, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 51, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 0, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 51, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 0, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 28, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 0, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [11, 2, 12, 0, 0, 0, 0, 3, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] Unknown
Crown events: [1, 0, 28, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 0, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 20, 0, 0, 0, 0, 0, 0] KeyPress { modifiers: 32 }
Crown events: [1, 0, 0, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 20, 0, 0, 0, 0, 0, 0] KeyPress { modifiers: 32 }
Crown events: [1, 20, 33, 0, 0, 0, 0, 0] KeyPress { modifiers: 32 }
Crown events: [1, 20, 0, 0, 0, 0, 0, 0] KeyPress { modifiers: 32 }
Crown events: [1, 0, 0, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, f, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 0, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 12, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 0, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, a, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
Crown events: [1, 0, 0, 0, 0, 0, 0, 0] KeyPress { modifiers: 0 }
App switch: 30952 /usr/bin/konsole
Mode events: EnableRatchet
App switch: 30952 /usr/bin/konsole
Mode events: EnableRatchet
Crown events: [11, 2, 12, 21, 2, 2, 32, 28, 19, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] Unknown
Crown events: [11, 2, 12, 21, 2, 2, 32, 28, 19, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] Unknown


```

</p>
</details>

**Comment by prefiks** [comment]
Thank you, could you please check if that version works better?

[program2.zip](https://github.com/prefiks/crown-controller/files/5279522/program2.zip)

**Comment by zkvsky** [comment]
There's no difference

**Comment by prefiks** [comment]
Oups, didn't update parsing for rotate event, one more try?
[program3.zip](https://github.com/prefiks/crown-controller/files/5279606/program3.zip)

**Comment by zkvsky** [comment]
It works now :D Thanks! Finally that damn Crown is useful under Linux

**Comment by zkvsky** [comment]
What about the `touch` event? is it possible to use that?
Global one doesn't seem to be working, the ones for audio

**Comment by prefiks** [comment]
Newly published version v0.2 added action for touch and release: 
```
touch:
  - KeyPress: "Space"
```

If you are using global from config.yaml from that repository, those only work when you have Ctrl key pressed (aka press Ctrl, rotate right and it will send key to switch to next song). You can change that by in changing Ctrl to None in global section (or you can add another section for none). I prefer it with Ctrl as it work even when active app have actions specified for crown.

**Comment by zkvsky** [comment]
Ctrl doesnt work for me, None does, which is what I wanted anyway. Everything else seems to be working so I'm closing.
